libraries/botbuilder-core/botbuilder/core/Teams/teams_activity_handler.py

```diff
@@ -0,0 +1,189 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+from http import HTTPStatus
+from botbuilder.schema import ActivityTypes, ChannelAccount
+from botbuilder.core.turn_context import TurnContext
+from botbuilder.core import ActivityHandler, MessageFactory, InvokeResponse
+from botbuilder.schema.teams import (
+    TeamInfo,
+    ChannelInfo,
+    TeamsChannelData,
+    TeamsChannelAccount,
+)
+from botframework.connector import Channels
+
+class TeamsActivityHandler(ActivityHandler):
+    async def on_turn(self, turn_context: TurnContext):
+        if turn_context is None:
+            raise TypeError("ActivityHandler.on_turn(): turn_context cannot be None.")
+
+        if hasattr(turn_context, "activity") and turn_context.activity is None:
+            raise TypeError(
+                "ActivityHandler.on_turn(): turn_context must have a non-None activity."
+            )
+
+        if (
+            hasattr(turn_context.activity, "type")
```

Comment:
I'm curious about the logic here: if the activity doesn't have a type attribute at all... should it pass? Because with this conditional is not gonna raise an error until it gets to line 34 and you try to access turn_context.activity.type

I believe that (and correct me if you had other intention) you're trying to validate that turn_context.activity.type exists and is not None, you can achieve that by 
`if not getattr(turn_context.activity, "type", None): ...(raise your error)`


I don't remember if you did a similar validation in your past PR but please take a look to make sure.

Reply:
Will resolve this in a different PR